src/swift_server.rs

```diff
@@ -756,14 +827,55 @@ impl ServerParams {
                     }
                 } else {
                     log::info!(target: "sim", "simulate tx (rpc): {:?}", SystemTime::now().duration_since(t1));
-                    Ok(SimulationStatus::SuccessRpc)
+                    sim_result.status = SimulationStatus::SuccessRpc;
+                    Ok(sim_result)
                 }
             }
             Ok(Err(err)) => {
                 log::warn!(target: "sim", "network sim error: {err:?}");
-                Ok(SimulationStatus::Degraded)
+                sim_result.status = SimulationStatus::Degraded;
+                Ok(sim_result)
+            }
+            Err(_) => {
+                sim_result.status = SimulationStatus::Timeout;
+                Ok(sim_result)
+            }
+        }
+    }
+
+    /// Simulate if auction params will be sanitized
+    async fn simulate_will_auction_params_sanitize(&self, order_params: &mut OrderParams) -> bool {
```

Comment:
can remove the `async` here since the client is subbed to the market and oracles.
these methods don't need the await:
`get_perp_market_account` => `try_get_perp_market_account`
`get_oracle_price_data_and_slot` => `try_get_oracle_price_data_and_slot`